smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/PaginationGenerator.java

```diff
@@ -154,13 +160,7 @@ private void writePager() {
                 });
 
                 writer.write("yield page;");
-                if (outputTokenName.contains(".")) {
-                    // Smithy allows one level indexing (ex. 'bucket.outputToken').
-                    String[] outputIndex = outputTokenName.split("\\.");
-                    writer.write("token = page[$S][$S];", outputIndex[0], outputIndex[1]);
-                } else {
-                    writer.write("token = page[$S];", outputTokenName);
-                }
+                writer.write("token = page$L;", destructurePath(outputTokenName));
```

Comment:
While verifying this fix with https://github.com/aws/aws-sdk-js-v3/pull/1580, I noticed that outputTokenName returns `"Marker"`, although smithy model contains `"EngineDefaults.Marker"`

Marking this PR as blocked. Feel free to dismiss this review if this issue is not reproducible.

Reply:
smithy model file used for verification https://github.com/trivikr/aws-sdk-js-v3/blob/update-models-10-16/codegen/sdk-codegen/aws-models/elasticache.2015-02-02.json

Reply:
done

Reply:
Fixed in [`6a1da87` (#223)](https://github.com/awslabs/smithy-typescript/pull/223/commits/6a1da874209da2f5569610d0b5f2d6f8ce509160)